src/verification_key.rs

```diff
@@ -194,10 +168,13 @@ impl VerificationKey {
                 .chain(msg),
         );
 
-        // We expect to recompute R as [s]B - [k]A = [k](-A) + [s]B.
-        let R = EdwardsPoint::vartime_double_scalar_mul_basepoint(&k, &self.minus_A, &s);
+        //       [8][s]B = [8]R + [8][k]A
+        // <=>   [8]R = [8][s]B - [8][k]A
+        // <=>   0 = [8](R - ([s]B - [k]A))
+        // <=>   0 = [8](R - R')  where R' = [s]B - [k]A
+        let R_prime = EdwardsPoint::vartime_double_scalar_mul_basepoint(&k, &self.minus_A, &s);
 
-        if R.compress().to_bytes() == signature.R_bytes {
+        if (R - R_prime).mul_by_cofactor().is_identity() {
```

Comment:
Why not do `is_small_order()`?

Reply:
No particular reason, I think I was trying to match the description in the spec more closely.